cmd/kubeless/function/function.go

```diff
@@ -245,11 +253,12 @@ func getFunctionDescription(funcName, ns, handler, file, deps, runtime, runtimeI
 
 	}
 
-	selectorLabels := map[string]string{}
-	for k, v := range funcLabels {
-		selectorLabels[k] = v
+	funcNodeSelectors := parseNodeSelectors(nodeSelectors)
+	if len(funcNodeSelectors) == 0 && len(defaultFunction.Spec.Deployment.Spec.Template.Spec.NodeSelector) != 0 {
+		funcNodeSelectors = defaultFunction.Spec.Deployment.Spec.Template.Spec.NodeSelector
 	}
-	selectorLabels["function"] = funcName
```

Comment:
mmh, I am confused about this removed code, was not being used?

Reply:
Yes, `selectorLabels` wasn't being used. I think it was somehow leftover from commit 47f38d5d3

Reply:
Earlier, `selectorLabels` was being used as the selector in the service spec. But, now, a default set of labels is being passed from `deploy.go` and together with other labels specified on the command line, they become the selector for the service spec.

https://github.com/kubeless/kubeless/blob/master/cmd/kubeless/function/deploy.go#L198-L202